README.md
# AeternumDev
The repository for the EU4 mod, Aeternum.


CHANGELOG.txt
############################################################




FOR THE LATEST PATCH NOTES SEE THE BOTTOM OF THIS DOCUMENT.




############################################################
VERSION 1.0.0 09/06/19

Gameplay changes: 

• 12 new full idea groups added.
• 21 new advisor types added for both western and asian cultures.
• Additional generic triggered modifiers.

Government changes:

• Maximum monarch points per monarch increased from 6/6/6 to 8/8/8.
• Minimum age to abdicate reduced from 60 to 40.
• Minimum length of rule to enable abdication reduced from 20 to 10.
• Legitimacy hit for abdication increased from -20 to -50.
• Prestige hit for abdication decreased from -50 to -20.
• Parliament debate duration decreased from 5 to 3.
• Prestige loss if parliament debate fails decreased from -20 to -10.

Diplomatic changes:

• Improve Relation speed decreased from 1 to 0.7.
• Abandon Union prestige hit increased from -25 to -40.
• Number of great powers decreased from 8 to 5.
• Development contribution from subjects to great power rank increased from 0.5 to 0.8.
• Maximum gold able to be taken in peace deal increased from 5 loans to 8.
• Share Map Prestige Cost reduced from -15 to -10.
• Number of possible rivals reduced from 3 to 2.
• Foreign province control penalty for IA reduced from -0.005 to -0.0025.
• Months before you can be challenged for Defender of the Faith set increased to 120 from 24.
• Monthly IA per month per free city increased to 0.008 from 0.005.
• Lack of electors IA penalty reduced from -0.1 to -0.05.
• HRE Heretics Penalty reduced from -0.01 to -0.008.

AI changes:

• AI development cap base increased from 10 to 20. 
• AI development cap multiplier increased from  to 3.
• AI scoring to give subsidies to nations opposing their rivals increased from 40 to 80.
• AI scoring for calling crusade on opposing nations increased from 40 to 80.
• AI scoring for excommunicating opposing nations increased from 40 to 50.
• AI scoring for taking on foreign debt increased from 0 to 20.
• Additonal AI scoring for taking on foreign debt for allies increased from 30 to 40.
• AI base aggressiveness increased from 200 to 300.
• AI aggressiveness when their enemy is perceived to be weak increased from 300 to 500.
• AI will want to send 60% of their military into enemy territory when in wars. Increased from 45%.
• AI army composition improved. 
• Artillery fraction in AI armies increased from 0.35 to 0.50.
• AI will more aggressively avoid going over their force limit.
• Pursue distance for AI armies chasing defeated enemies increased from 100 to 150.
• The AI will want to buy 1 general per 30 regiments. Increased from 15.
• The AI will want to buy 1 admiral per 25 ships. Increased from 10.
• AI desire for releasing a nation in a peace deal if they are the same culture group multiplied by 0.85. Increased from 0.65.
• AI desire for releasing a nation in a peace deal if they are the same culture group (not same culture) multiplied by 0.85. Increased from 0.65.
• AI desire for releasing a nation if they are both HRE members increased from 2.0 to 4.0.
• AI desire for demanding war reparations in a peace deal increased from 0.1 to 0.2.
• AI desire to declare a crusade against a rival increased from 10 to 20.
• AI desire to declare a crusade against a rival neighbour increased from 25 to 30.
• AI reluctance to fire advisors decreased from advisors level 3 to level 2.
• AI cap on saving ducats increased from 60 months of income to 100.
• Monthly chance of AI picking a new mission increased from 33 to 66.

Combat changes:

• Maximum war exhaustion increased from 20 to 30.
• Conquistador cost decreased from 50 to 30.
• Explorer cost decreased from 50 to 30.
• Artillery barrage cost decreased from 50 to 30.
• Naval barrage cost decreased from 50 to 25.
• Replace rival cost decreased from 100 to 50.
• Loot devastation impact increased from 10 to 15.
• Scorched earth devastation impact increased from 5 to 15.
• Rebel leader base power increased from 20 to 25.
• Sea landing penalty increased from -2 to -3.
• Base siege attrition reduced from 1.00 to 0.95.

Graphical changes: 

• Minor replacements to key flags.
• Border width reduced.
• Maximum flags displayed beneath a trade node increased from 5 to 10.
• Trade routes on political map displayed more clearly.
• Maximum zoom out display for trade routes increased from 400 to 800.

Development Changes:

• Upon reaching certain levels of development provinces will gain certain bonuses displated as a province modifier.

National Mechanics changes:

• Move capital cost recreased from 200 to 150.
• Duration of unrest after removing an accepted culture increased from 5 years to 20.
• Overextension potency reduced from 1.0 to 0.8.
• Overseas overextension potency decreased from 0.5 to 0.4.

Unrest changes:

• Revolt base size increased from 4 to 5.

Estate changes:

• Minimum base decay of estate loyalty decreased from 1.0 to 0.5 per year.
• Maximum base decay of estate loyalty decreased from 2.0 to 1.0 per year.

Espionage Changes:

• Infiltrate Administration cost reduced from 40 Spy Network size to 25.
• Sabotage Reputation cost reduced from 60 Spy Network size to 50.
• Steal Maps cost reduced from 50 to 35.
• Support Rebels cost reduced from 60 to 45.
• Sow Discontent cost reduced from 80 to 70.
• Agitate For Liberty cost reduced from 90 to 70.
• Sabotage Recruitment cost increased from 80 to 90.
• Slander Merchants cost decreased from 70 to 65.

Peace Deal Changes:

• Concede Province cost reduced from 1.0 to 0.8
• Cost to demand capital multiplier increased from 1.2 to 1.5.
• Cost to take land from non co-belligerants increased from 2 to 3.

############################################################
############################################################

common/defines/00_aeternum.lua
-- Aeternum defines changes

--NDiplomacy

NDefines.NDiplomacy.SHAREMAP_PRESTIGE_TRANSFER_LOSE = -10 --Reduced from -15
NDefines.NDiplomacy.SHAREMAP_PRESTIGE_TRANSFER_GAIN = 10 --Same as vanilla
NDefines.NDiplomacy.NUM_POSSIBLE_RIVALS = 2 --reduced from 3
NDefines.NDiplomacy.HRE_FOREIGN_CONTROL_PENALTY = -0.0025 --reduced from 0.005 to encourage a stronger HRE
NDefines.NDiplomacy.DEFENDER_OF_FAITH_MONTHS = 120 --Number of months you have DoF for before it can be taken away by another nation. Increased from 24.
NDefines.NDiplomacy.IMPERIAL_CITY_IA = 0.008 -- Monthly per imperial free city. Increased from 0.005
NDefines.NDiplomacy.REMOVE_FREECITY_INFLUENCE_COST = 1 --Decreased from 5
NDefines.NDiplomacy.LACK_OF_ELECTORS_HIT = -0.05 -- Also applied to vassalized electors. Decreased from 0.1
NDefines.NDiplomacy.HRE_HERETIC_PENALTY = -0.008 --Decreased from 0.01
NDefines.NDiplomacy.INFILTRATE_ADMINISTRATION_COST = 25 -- Reduced from 40
NDefines.NDiplomacy.SABOTAGE_REPUTATION_COST = 50 -- Reduced from 60
NDefines.NDiplomacy.STEAL_MAPS_COST = 35 --Reduced from 50
NDefines.NDiplomacy.SUPPORT_REBELS_COST = 45 --Reduced from 60
NDefines.NDiplomacy.SOW_DISCONTENT_COST = 70 -- Reduced from 80
NDefines.NDiplomacy.AGITATE_FOR_LIBERTY_COST = 70 -- Reduced from 90
NDefines.NDiplomacy.SABOTAGE_RECRUITMENT_COST = 90 --Increased from 80
NDefines.NDiplomacy.SLANDER_MERCHANTS_COST = 65 --Decreased from 70
NDefines.NDiplomacy.PEACE_COST_CONCEDE_PROVINCE = 0.8 --The cost to annex colonial provinces. Changed from 1.
NDefines.NDiplomacy.PEACE_COST_DEMAND_CAPITAL_MULT = 1.5 --The cost multiplier to demand a war-participants capital. Increased from 1.2
NDefines.NDiplomacy.ALLY_PEACE_COST_MULT = 3 --The cost of taking land from non co-belligerants. Increased from 2.
NDefines.NDiplomacy.IMPROVE_RELATION_SPEED = 0.7 --Decreased from 1.
NDefines.NDiplomacy.ABANDON_UNION_PRESTIGE = -40 --Presige loss from abandoning a PU. Increased from -25
NDefines.NDiplomacy.NUM_OF_GREAT_POWERS = 5 --Decreased from 8
NDefines.NDiplomacy.LEAVING_GREAT_POWER_YEARS = 10 --Increased from 5
NDefines.NDiplomacy.GREAT_POWER_SUBJECT_CONTRIBUTION = 0.8 --Amount your subjects contribute to your great power rank. Increased from 0.5.
NDefines.NDiplomacy.PEACE_COST_GOLD_MAX = 8 --Increased from 5

--NCountry

--NCountry
NDefines.NCountry.MONARCH_MAX_SKILL = 8 --Highest amount of mana a leader can have. Increased from 6.
NDefines.NCountry.ABDICATE_AGE_THRESHOLD = 40 --Decreased from 60
NDefines.NCountry.ABDICATE_RULING_LENGTH_THRESHOLD = 10 --Decreased from 20
NDefines.NCountry.ABDICATE_LEGITIMACY_HIT = -50 --Increased from -20
NDefines.NCountry.ABDICATE_PRESTIGE_HIT = -20 --Decreased from -50
NDefines.NCountry.REVOLT_SIZE_BASE = 5 --Increased from 4.
NDefines.NCountry.STATE_MAINTENANCE_CULTURE_FACTOR = 0.33 --Increased from 0.25
NDefines.NCountry.WAREXHAUSTION_REDUCTION = 2.5 --Increased from 2 due to higher W.E cap
NDefines.NCountry.STATE_MAINTENANCE_CULTURE_FACTOR = 0.30		-- Increased from 0.25
NDefines.NCountry.ADVISOR_COST_INCREASE_PER_YEAR = 0.006 --Increased from 0.005
NDefines.NCountry.PROMOTE_COST_MONTHS_ADVISOR_SALARY = 70.0 --Increased from 60.0
NDefines.NCountry.MERCHANT_REPUBLIC_SIZE_LIMIT = 22 --Increased from 20
--Changes to estates. These should make estate decisions more meaningful and the effects more important.
NDefines.NCountry.ESTATE_LOYALTY_DECAY_BASE_MIN = 0.5		-- Decreased from 1.0 Loyalty change per year at middle level. 
NDefines.NCountry.ESTATE_LOYALTY_DECAY_BASE_MAX = 1.0		-- Decreased from 2.0 Loyalty change per year at most extreme levels.
NDefines.NCountry.PARLIAMENT_DEBATE_DURATION = 3 --Decreased from 5
NDefines.NCountry.PARLIAMENT_PRESTIGE_HIT = -10 --Decreased from 20. Prestige loss if debate fails.
NDefines.NCountry.MAX_WAR_EXHAUSTION = 30 --Increased from 20
NDefines.NCountry.PS_BUY_CONQUISTADOR = 30 --Decreased from 50
NDefines.NCountry.PS_BUY_EXPLORER = 30 --Decreased from 50
NDefines.NCountry.PS_ARTILLERY_BARRAGE = 30 -- Decreased from 50
NDefines.NCountry.PS_NAVAL_BARRAGE = 25 --Decreased from 50
NDefines.NCountry.PS_REPLACE_RIVAL = 50 --Decreased from 100
NDefines.NCountry.PS_MOVE_CAPITAL = 150 --Decreased from 200
NDefines.NCountry.PS_REMOVE_ACCEPTED_CULTURE_UNREST_DURATION = 20 --Increased from 5
NDefines.NCountry.PS_REMOVE_ACCEPTED_CULTURE = 25 --Cost to remove accepted culture. Increased from 10
NDefines.NCountry.OVEREXTENSION_FACTOR = 0.8 --Decreased from 1.0, overextension is less potent. Fewer rebels, more dangerous when they do spawn.
NDefines.NCountry.OVEREXTENSION_OVERSEAS_FACTOR = 0.4 --Decreased fron 0.5, Fewer rebels, more dangerous when they do spawn.
NDefines.NCountry.DEVELOPMENT_CAP_BASE = 20	-- Increased from 10. AI will not develop provinces that have more development than this or DEVELOPMENT_CAP_MULT*original development (whichever is bigger)
NDefines.NCountry.DEVELOPMENT_CAP_MULT = 3 -- Increased from 2
NDefines.NCountry.WAREXHAUSTION_REDUCTION = 3 --Increased from 2
NDefines.NCountry.ADVISOR_COST_INCREASE_PER_YEAR = 0.006 --Increased from 0.005
NDefines.NCountry.PROMOTE_COST_MONTHS_ADVISOR_SALARY = 70.0 --Increased from 60.0
NDefines.NCountry.MERCHANT_REPUBLIC_SIZE_LIMIT = 22 --Increased from 20
NDefines.NCountry.PS_REMOVE_ACCEPTED_CULTURE = 25 --Increased from 10
NDefines.NCountry.LIBERTY_DESIRE_HISTORICAL_RIVAL = 40 --Decreased from 50
NDefines.NCountry.LIBERTY_DESIRE_GREAT_POWER_VASSAL = 55 --Increased from 50
NDefines.NCountry.LOST_INDEPENDANCE_WAR_LIBERTY_DESIRE = -65 --Increased from -50
NDefines.NCountry.MAX_CROWN_COLONIES = 6 --Increased from 4


--NEconomy
NDefines.NEconomy.DEBASE_MAX_STORED_MONTHS = 30 --Decreased from 30
NDefines.NEconomy.DEBASE_MONTHS_PER_CHARGE = 6 --Decreased from 12
NDefines.NEconomy.DEBASE_MAX_CORRUPTION = 6 --Decreased from 90
NDefines.NEconomy.DEBASE_ADDED_CORRUPTION = 2 --Decreased from 2
NDefines.NEconomy.AUTONOMY_AT_DIPLO_ANNEX = 50 --Decreased from 60
NDefines.NEconomy.AUTONOMY_AT_CONQUEST = 60 --Increased from 50
NDefines.NEconomy.AUTONOMY_AT_CONQUEST_CORE = 10 --Increased from 0
NDefines.NEconomy.BANKRUPTCY_PROVINCE_DEVASTATION_GAIN = 15 --Increased from 10
NDefines.NEconomy.MAX_PROVINCE_SELL_PRICE = 200 --Increased from 100
NDefines.NEconomy.MERCHANT_SPEED = 15.0 --Decreased from 20.0
NDefines.NEconomy.EMBARGO_BASE_EFFICIENCY = 0.33 --Decreased from 50
NDefines.NEconomy.EMBARGO_MERCANTILISM_EFFICIENCY = 33 --Decreased from 50
NDefines.NEconomy.REGAIN_COST_BEFORE_PROGRESS = 0.15 --Increased from 0.10
NDefines.NEconomy.COT_DOWNGRADE_PRESTIGE_COST = -20 --Increased from 10

--Changes to vassals
NDefines.NCountry.LIBERTY_DESIRE_HISTORICAL_RIVAL = 40 --Decreased from 50.
NDefines.NCountry.LIBERTY_DESIRE_GREAT_POWER_VASSAL = 55 --Increased from 50.
NDefines.NCountry.LOST_INDEPENDANCE_WAR_LIBERTY_DESIRE = -65 --Increased from -50.
NDefines.NCountry.MAX_CROWN_COLONIES = 6 --Increased from 4.
NDefines.NCountry.CAN_CONVERT_TERRITORY_CULTURE = 0 --No longer able to culture convert territory provinces.


--NEconomy
-- Changes to the economy
NDefines.NEconomy.DEBASE_ADDED_CORRUPTION = 2.5 --Increased from 2.
NDefines.NEconomy.AUTONOMY_AT_DIPLO_ANNEX = 30 --Decreased from 60.
NDefines.NEconomy.AUTONOMY_AT_CONQUEST = 75 --Increased from 50.
NDefines.NEconomy.AUTONOMY_AT_CONQUEST_CORE = 10 --Increased from 0.
NDefines.NEconomy.BANKRUPTCY_PROVINCE_DEVASTATION_GAIN = 15 --Increased from 10.
NDefines.NEconomy.MAX_PROVINCE_SELL_PRICE = 200 --Increased from 100.
NDefines.NEconomy.MERCHANT_SPEED = 15.0 --Decreased from 20.
NDefines.NEconomy.EMBARGO_BASE_EFFICIENCY = 0.33 --Decreased from 0.50.
NDefines.NEconomy.EMBARGO_MERCANTILISM_EFFICIENCY = 33 --Decrased from 0.50.
NDefines.NEconomy.REGAIN_COST_BEFORE_PROGRESS = 0.15 --Increased from 0.10.
NDefines.NEconomy.COT_DOWNGRADE_PRESTIGE_COST = -20 --Increased from -10.


--NMilitary
NDefines.NMilitary.LOOT_DEVASTATION_IMPACT = 15 --Increased from 10
NDefines.NMilitary.SCORCHED_DEVASTATION_IMPACT = 15 --Increased from 5
NDefines.NMilitary.SEA_LANDING_PENALTY = -3 --Increased from -2
NDefines.NMilitary.REBEL_LEADER_POWER = 25 --Increased from 20. Fewer rebels, more dangerous when they do spawn.
NDefines.NMilitary.SIEGE_ATTRITION = 0.80 -- Decreased from 100
NDefines.NMilitary.SIEGE_DISEASE_IMPACT = 0.07 --Increased from 0.05
NDefines.NMilitary.NATIVE_FEROCITY_IMPACT = 0.06 --Increased from 0.05
NDefines.NMilitary.SLACKEN_AP_DROP = 0.075 --Increased from 0.05, 50% increase
NDefines.NMilitary.LACKEN_MANPOWER_INCREASE = 3.0 --50% increase
NDefines.NMilitary.ASSAULT_ATTACKER_LOSS = 0.75 --Decreased from 1.0
NDefines.NMilitary.ASSAULT_DEFENDER_LOSS = 0.10 --Increased from 0.05
NDefines.NMilitary.CAVALRY_SPEED = 0.75 --Increased from 0.7
NDefines.NMilitary.LOOTED_MAX = 6 --Increased from 5
NDefines.NMilitary.DEFAULT_WARGOAL_WARSCORE_BONUS	= 7 --Increased from 5
NDefines.NMilitary.COMBAT_DICE_SIDE = 9 --Decreased from 10
NDefines.NMilitary.FORT_FLIPPING_TIME = 60 --Increased from 30
NDefines.NMilitary.SIEGE_DISEASE_IMPACT = 0.08 --Increased from 0.05
NDefines.NMilitary.ASSAULT_DEFENDER_LOSS = 0.12 --Increased from 0.05
NDefines.NMilitary.CAVALRY_SPEED = 0.8 --Increased from 0.7
NDefines.NMilitary.LOOTED_MAX = 6 --Increased from 5.
NDefines.NMilitary.DEFAULT_WARGOAL_WARSCORE_BONUS	= 8 --Increased from 5.
NDefines.NMilitary.COMBAT_DICE_SIDE = 6 --Reduced from 10
NDefines.NMilitary.FORT_FLIPPING_TIME = 60 --Increased from 30.


--NAI
NDefines.NAI.DIPLOMATIC_ACTION_SUBSIDIES_POWERBALANCE_FACTOR = 60 --AI scoring to give subsidies to nations blocking/fighting power balance threat. Increased from 40.
NDefines.NAI.DIPLOMATIC_ACTION_CRUSADE_POWERBALANCE_FACTOR = 80 --AI scoring for calling crusade on power balance threat. Increased from 40.
NDefines.NAI.DIPLOMATIC_ACTION_EXCOMMUNICATE_POWERBALANCE_FACTOR = 50 --AI scoring for excommunicating power balance threat. --Increased from 40
NDefines.NAI.DIPLOMATIC_ACTION_TAKE_ON_DEBT_BASE_FACTOR = 20 -- AI base scoring for Take on Foreign Debt (needs to be ally or block rival/power balance threat to even apply). Increased from 0.
NDefines.NAI.DIPLOMATIC_ACTION_TAKE_ON_DEBT_ALLY_FACTOR = 40 --AI additional scoring for Take on Foreign Debt with nations we are allied to. Increased from 30.
NDefines.NAI.DIPLOMATIC_ACTION_ENFORCE_PEACE_FORCE_BALANCE_MULT = 1.20 --Increased from 1.
NDefines.NAI.DIPLOMATIC_ACTION_SUBSIDIES_MAX_OCCUPIED = 0.55 --Increased from 0.50
NDefines.NAI.AGGRESSIVENESS = 300 --Increased from 200. 
NDefines.NAI.AGGRESSIVENESS_BONUS_EASY_WAR = 500 --Increased from 300
NDefines.NAI.INVADING_MAX_AWAY_RATIO = 0.60 --Increased from 0.45
NDefines.NAI.ARTILLERY_FRACTION = 0.40 --Increased from 0.35
NDefines.NAI.OVER_FORCELIMIT_AVOIDANCE_FACTOR = 20 --Increased from 10
NDefines.NAI.PURSUE_DISTANCE = 150 --Increased from 100
NDefines.NAI.REGIMENTS_PER_GENERAL = 30 --Increased from 15. This was increased after extensive testing saw the AI regularly exceeding their leader limit by upwards of 5.
NDefines.NAI.MIN_SHIPS_FOR_ADMIRAL = 25 -- Increased from 10. This was increased after extensive testing saw the AI regularly exceeding their leader limit by upwards of 5.
NDefines.NAI.PEACE_TERMS_RELEASE_VASSAL_SAME_CULTURE_MULT = 0.85 -- AI desire for releasing a country is multiplied by this if they are the same culture group as releaser. Increased from 0.65
NDefines.NAI.PEACE_TERMS_RELEASE_VASSAL_SAME_CULTURE_GROUP_MULT = 0.90 -- AI desire for releasing a country is multiplied by this if they are the same culture group (but not same culture) as releaser. Increased from 0.75
NDefines.NAI.PEACE_TERMS_RELEASE_ANNEXED_HRE_MULT = 4.0 -- AI desire for releasing a country is multiplied by this if both are HRE members --Increased from 2.0
NDefines.NAI.PEACE_TERMS_WAR_REPARATIONS_BASE_MULT = 0.2 -- AI desire for war reparations through peace. Increased from 0.l
NDefines.NAI.DIPLOMATIC_ACTION_CRUSADE_RIVAL_FACTOR = 20 -- AI scoring for crusade is increased by this if they are rivals. Increased from 10.
NDefines.NAI.DIPLOMATIC_ACTION_CRUSADE_NEIGHBOUR_FACTOR = 30 -- AI scoring for crusade is increased by this if they are neighbours. Increased from 25.
NDefines.NAI.ADVISOR_MIN_SKILL_RELUCTANT_FIRE = 2 --Decreased from 3
NDefines.NAI.MAX_SAVINGS = 100 --Increased from 60. AI will keep a maximum of this * their monthly income in long-term savings.
NDefines.NAI.MISSION_PICK_CHANCE = 66 -- Increased from 33. Monthly chance of AI picking a mission if it lacks one (of 100)
NDefines.NAI.PEACE_HIGH_WAR_EXHAUSTION_THRESHOLD = 15 --Increased from 10
NDefines.NAI.ACCEPTABLE_BALANCE_DEFAULT = 1.6 --Decreased from 1.7
NDefines.NAI.ACCEPTABLE_BALANCE_FRIEND_IN_COMBAT = 0.70 --Decreased from 0.75
NDefines.NAI.DEVELOP_IN_SUBJECTS_PREFERENCE = 0.65 --Decreased from 0.75
NDefines.NAI.CALL_IN_ALLIES_POWER_RATIO = 3.5 --Decreased from 4.0

--NGraphics
NDefines.NGraphics.MAX_TRADE_NODE_FLAGS_SHOWN = 10 --Increased from 5
NDefines.NGraphics.DRAW_TRADEROUTES_CUTOFF = 800 --Increased from 400

NDefines.NNationDesigner.RULER_MAX_SKILL = 8 --Increased from 6

NDefines.NFrontend.CAMERA_END_X = 4958.0